src/parquet/metadata.h

```diff
@@ -171,7 +190,7 @@ class PARQUET_EXPORT FileMetaData {
 
   const ApplicationVersion& writer_version() const;
 
-  void WriteTo(OutputStream* dst) const;
+  void WriteTo(OutputStream* dst, EncryptionProperties* encryption = nullptr) const;
```

Comment:
Why is this needed both here and in the constructor?

Reply:
constructor is for decryption while reading and here is for encryption while writing